generators/upgrade/index.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+var util = require('util'),
+    generators = require('yeoman-generator'),
+    chalk = require('chalk'),
+    scriptBase = require('../generator-base'),
+    packagejs = require('../../package.json'),
+    fs = require('fs'),
+    shelljs = require('shelljs'),
+    exec = require('child_process').exec,
+    semver = require('semver'),
+    Git = require('git-wrapper');
+
+var UpgradeGenerator = generators.Base.extend({});
+
+util.inherits(UpgradeGenerator, scriptBase);
+
+var configOptions = {};
+
+/* Constants used throughout */
+const GENERATOR_JHIPSTER = 'generator-jhipster',
+    UPGRADE_BRANCH = 'jhipster_upgrade';
+
+module.exports = UpgradeGenerator.extend({
+    constructor: function () {
+        generators.Base.apply(this, arguments);
+
+        this.currentVersion = packagejs.version;
+        this.logo = configOptions.logo;
+        this.git = new Git();
+
+        this.gitCheckout = function(branch) {
+            var done = this.async();
+            this.git.exec('checkout', {q: true}, [branch], function(err, msg) {
+                if (err != null) this.error('Unable to checkout branch ' + branch + ':\n' + err);
+                this.log('Checked out branch \'' + branch + '\'');
+                done();
+            }.bind(this));
+        };
+
+        this.gitCommitAll = function(commitMsg, callback) {
+            var commit = function() {
+                this.git.exec('commit', {q: true}, ['-m \"' + commitMsg + '\"', '-a', '--allow-empty'], function(err, msg) {
+                    if (err != null) return this.error('Unable to commit in git:\n' + err);
+                    this.log('Committed: ' + commitMsg);
+                    callback();
+                }.bind(this));
+            }.bind(this);
+            this.git.exec('add', {}, ['-A'], function(err, msg) {
+                if (err != null) return this.error('Unable to add resources in git:\n' + err);
+                commit();
+            }.bind(this));
+        };
+
+        // This adds support for a `--force` flag
+        this.option('force', {
```

Comment:
I think --force is already available in yeoman so need to define again here. You can use it directly using `this.options['force']`